File: solutions/javascript/pangram/1/pangram.js
//
// This is only a SKELETON file for the 'Pangram' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const isPangram = (sentence) => {
  const lowerSentence = sentence.toLowerCase();

  const letterSet = new Set();

  for (let char of lowerSentence) {
    if (char >= 'a' && char <= 'z') {
      letterSet.add(char);
    }
  }

  return letterSet.size === 26;
};

File: solutions/python/pangram/1/pangram.py
def is_pangram(sentence):
    sentence = sentence.lower()
    cnt = [0] * 26
    for c in sentence:
        if 'a' <= c <= 'z':
            cnt[ord(c) - ord('a')] += 1

    for n in cnt:
        if n == 0:
            return False
    return True
